xalloc.h

```diff
@@ -0,0 +1,80 @@
+/*
+   xalloc.h -- malloc and related fuctions with out of memory checking
+   Copyright (C) 2014 Guus Sliepen <guus@meshlink.io>
+
+   This program is free software; you can redistribute it and/or modify
+   it under the terms of the GNU General Public License as published by
+   the Free Software Foundation; either version 2, or (at your option)
+   any later version.
+
+   This program is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+   GNU General Public License for more details.
+
+   You should have received a copy of the GNU General Public License along
+   with this program; if not, write to the Free Software Foundation, Inc., Foundation,
+   51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+*/
+
+#ifndef __MESHLINK_XALLOC_H__
+#define __MESHLINK_XALLOC_H__
```

Comment:
wasn't clear whether we want to have that list and xalloc implementation copied over from meshlink at all or whether we'd maybe prefer to add a common dependency?